### ISSUE TITLE
Documentation for callbacks: how to get the api object

### ISSUE THREAD

**Issue by shark8me** [tflearn/tflearn#646]
Hi all,
The [documentation](http://tflearn.org/getting_started/#training-callbacks) for adding callback methods describes how to create the object and attach it to the model.fit method. 

```python
monitorCallback = MonitorCallback(api) # "api" is your API class
model = ...
model.fit(..., callbacks=monitorCallback)
```
However, it is not clear how to get a handle to the *api* object. I did not find any examples of the same in the code either. 
Could you please add this to the documentation?

Thanks!

regards
Kiran

**Comment by XucongZhang** [comment]
I finally figured it out.
You first need to define your own callback function:
```
class MonitorCallback(tflearn.callbacks.Callback):
    def __init__(self):
        self.train_losses = []
    def on_batch_end(self, training_state, snapshot=False):
    	print("The training loss is: ", training_state.global_loss)
    	self.train_losses.append(training_state.global_loss)
```
And then you just need to initialize this class:
```
monitorCallback = MonitorCallback()
model = ...
model.fit(..., callbacks=monitorCallback)
```

The train loss will be stored in `monitorCallback.train_losses` now.

I use `on_batch_end` here, but you can also use other functions.
